ElCoditoBandito/vest-hive-test
# Create Core UI Components and Layout #6

This PR implements the core UI components and layout structure for the VestHive platform.

### Components Added
1. Layout Components:
   - Root layout with navigation
   - Responsive sidebar
   - Mobile navigation

2. Dashboard Components:
   - Performance overview chart
   - Activity feed
   - Quick actions
   - Metrics cards

3. Club/Team Components:
   - Club and team cards
   - Member list component
   - Performance tracking charts
   - Creation dialogs

4. Form Components:
   - Input fields
   - Text areas
   - Labels
   - Dialogs

5. Performance Tracking:
   - Performance charts
   - Performance tables
   - Data formatting utilities

6. Shared UI Components:
   - Button variants
   - Card components
   - Table components
   - Progress indicators

### Technical Details
- Uses shadcn/ui components for consistency
- Implements responsive design
- Follows Tailwind best practices
- Type-safe with TypeScript
- Reusable component architecture

### Testing Instructions
1. Run the development server
2. Check responsive behavior
3. Test form validations
4. Verify component styling

### Notes
- All components are built using only Tailwind's core utility classes
- Components are modular and can be reused across the application
- Form validation is implemented on required fields
- Performance charts use Recharts library

Closes #6
